### PR TITLE
stop redirecting developers to details step when there are no versions.

### LINKED ISSUE #4167: an incomplete addon causes a redirect loop
If an add-on has all STATUS_DISABLED (e.g. rejected) versions and is STATUS_NULL (Incomplete) then any attempt to access it's devhub edit pages gets stuck in a redirect loop from `/submit/details/` to `/submit/finish/'.

I think this is a regression from https://github.com/mozilla/addons-server/pull/4073, in particular the changes to how [`has_required_metadata`](https://github.com/mozilla/addons-server/pull/4073/files#diff-d5d4f5e227c5a588706b9e9f6879197eR1260) works.

### PR DESCRIPTION
**Pull request by eviljeff**
fixes #4167 
(I hope!  The edge cases were making my brain hurt)

### CODE REVIEW

File: src/olympia/addons/tests/test_models.py [hunk 1]
```diff
@@ -1456,6 +1456,57 @@ def test_unlisted_has_complete_metadata(self):
...
+        delete_translation(addon, 'summary')
+        addon = Addon.objects.get(id=3615)
+        assert not addon.has_complete_metadata()
+
+        for status in amo.STATUS_CHOICES_ADDON:
+            addon.update(status=status)
+            if status == amo.STATUS_NULL:
+                assert addon.should_redirect_to_submit_flow()
+            else:
+                assert not addon.should_redirect_to_submit_flow()
```
Comment (EnTeQuAk) [inline src/olympia/addons/tests/test_models.py]: I personally would love to un-loop this code and rather have 5 addon.update() statements with their explicit assert statements than a for loop that can break in so many ways (or, even worse, won't break) if things change (e.g amo.STATUS_NULL get's removed from STATUS_CHOICES_ADDON)

Comment (eviljeff) [inline src/olympia/addons/tests/test_models.py]: It's a double edged sword - if we break out `STATUS_CHOICES_ADDON` then if a _new_ status is added to it then it won't be tested and it probably should be.  Perhaps I should drop `STATUS_NULL` from a list of `STATUS_CHOICES_ADDON` keys and test the rest; then test `STATUS_NULL` explicitly after.

Comment (EnTeQuAk) [inline src/olympia/addons/tests/test_models.py]: indeed it should be. I can live with both for now, we really need some parametrize stuff in our test-suite :(

Comment (eviljeff) [inline src/olympia/addons/tests/test_models.py]: https://github.com/mozilla/addons-server/pull/4171/commits/44a7bec0aed39d19d1bdcc6d6a43edb4d539ba6b
